api/utils/dependencies.py

```diff
@@ -22,14 +22,14 @@ def get_current_user(db: Session = Depends(get_db), token: str = Depends(oauth2_
         headers={"WWW-Authenticate": "Bearer"},
     )
     try:
-        payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
-        username: str = payload.get("username")
-        if username is None:
+        payload = jwt.decode(token, settings.SECRET_KEY, algorithms=[settings.ALGORITHM])
+        user_id: str = payload.get("user_id")
+        print(user_id)
+        if user_id is None:
             raise credentials_exception
-        token_data = TokenData(username=username)
-    except JWTError:
+    except (JWTError, jwt.exceptions.ExpiredSignatureError, jwt.exceptions.DecodeError):
         raise credentials_exception
-    user = db.query(User).filter(User.username == token_data.username).first()
+    user = db.query(User).filter(User.id == user_id).first()
```

Comment:
is this approach approved by others?

Reply:
The access token created in the routes/auth.py file had always been encoded using the `user_id`. 

The `create_access_token` method of the `UserService`, class takes a `user_id` string as one of its arguments.